test/unit/Command/AssertBackwardsCompatibleTest.php

```diff
@@ -280,7 +280,7 @@ public function testExecuteWithDefaultRevisionsNotProvided() : void
     {
         $fromSha       = sha1('fromRevision', false);
         $toSha         = sha1('toRevision', false);
-        $versions      = VersionsCollection::fromArray(['1.0.0', '1.0.1']);
+        $versions      = new VersionsCollection(Version::fromString('1.0.0'), Version::fromString('1.0.1'));
```

Comment:
@ntzm I see that this change caused some difficulties, because the ability to create the collection from version strings was very handy I would say now that I'm seeing use cases. Reasoning on my side was making things strict as possible, in order to avoid any ambiguity and simplify validation. So between `string` and `Version` inputs I decided to allow passing `Version` instances only.

I might consider introducing `fromVersionStrings()` and `fromArray()` named constructors for `3.1.0`. Of course, feel free to [suggest features](https://github.com/nikolaposa/version/issues) you might find useful.

Reply:
I personally think it's a good change 👍